src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VtableIndexStubGenerator.cs

```diff
@@ -358,20 +408,93 @@ private static IMarshallingGeneratorFactory GetMarshallingGeneratorFactory(StubE
 
             return (
                 methodStub.ContainingSyntaxContext.AddContainingSyntax(
-                    new ContainingSyntax(
-                        TokenList(Token(SyntaxKind.InternalKeyword), Token(SyntaxKind.PartialKeyword)),
-                        SyntaxKind.InterfaceDeclaration,
-                        Identifier("Native"),
-                        null))
+                    NativeTypeContainingSyntax)
                 .WrapMemberInContainingSyntaxWithUnsafeModifier(
                     PrintGeneratedSource(
                         methodStub.StubMethodSyntaxTemplate,
-                        methodStub.ContainingSyntaxContext.ContainingSyntax[0],
                         methodStub.SignatureContext,
-                        code)),
+                        code)
+                    .WithExplicitInterfaceSpecifier(ExplicitInterfaceSpecifier(IdentifierName(methodStub.ContainingSyntaxContext.ContainingSyntax[0].Identifier)))),
+                methodStub.Diagnostics.Array.AddRange(diagnostics.Diagnostics));
+        }
+
+        private const string ThisParameterIdentifier = "@this";
```

Comment:
I really would prefer not to see escaped keywords in our generated code. Let's go with `inst` or `thisPtr` or some other mechanism. We shouldn't need to rely on escaped keywords.

Reply:
I was worried that `inst` or `thisPtr` was more likely to conflict with user-defined identifiers than `@this`, so I went with `@this` so I could avoid having to add the extra underscores and make the code look a little cleaner. I'll change it.

Reply:
> I went with @this so I could avoid having to add the extra underscores and make the code look a little cleaner

I see. That is an annoying problem. Sigh. I'll defer to you here, if this makes the codegen easier with less ceremony let's go with it.